integr8ly/integreatly-operator
# updating metrics queries to work with newer version of kube state met…

…rics

# Issue link
https://issues.redhat.com/browse/MGDAPI-2634

# What
Changes the following metrics
- `kube_pod_container_resource_limits_memory_bytes` to  `kube_pod_container_resource_requests{resource='memory'}`
- `kube_node_status_capacity_memory_bytes` to `kube_node_status_capacity{resource='memory'}`
- `kube_pod_container_resource_requests_memory_bytes` to `kube_pod_container_resource_requests{resource=memory}`
- `kube_node_status_allocatable_memory_bytes` to `kube_node_status_allocatable{resource='memory'}`

- `kube_node_status_allocatable_cpu_cores` to `kube_node_status_allocatable{resource='cpu'}`
- `kube_pod_container_resource_requests_cpu_cores` to `kube_node_status_allocatable{resource='cpu'}`
- `kube_pod_container_resource_limits_cpu_cores` to `kube_pod_container_resource_limits{resource='cpu'}`

- `kube_node_labels{label_node_role_kubernetes_io_compute='true'}` to `kube_node_role{role='worker'}`

- Removed `by (namespace)` in case of single stat
- Renived `created_by_name` from `kube_pod_info` metric queries

Changes include updates to the following 
Grafana Dashboards
- Cluster Resource
- Resource By Namespace
- Resource By Pod

Prometheus Alert
- AMQOnlineContainerHighMemory 
- ClusterSchedulableMemoryLow
- ClusterSchedulableCPULow
- MultiAZPodDistribution
- ThreeScaleContainerHighMemory
- ThreeScaleContainerHighCPU

Scripts
- capture_resource_metrics.sh




# Verification steps
Verification is required on OpenShift 4.8 and OpenShift pre 4.7.21

4.8 cluster prepared titled `lfitzger-g` in ocm
Pre 4.7.21 cluster is a 4.7.6 cluster titled `lf-476` in ocm

In each of the two clusters provided

Grafana Dashboards
- Cluster Resource
- Resource By Namespace
- Resource By Pod

Each Dashboard should have full metrics expected, e.g. in each memory and cpu table there should appear 5 columns (actual, requested, %actual against requested, limits, %actual against limits) - note deploy pods in 3scale namespace wont return any data, please select another pod.
there should be no single single stat or graphs with no data or N/A or error saying `Only queries that return single series/table is supported`

Alerts
- AMQOnlineContainerHighMemory - only relevant to RHMI which makes me think we should cherry-pick these changes to the 2.9 branch so they could get picked up also in any future RHMI releaes @matskiv wdyt?
- ClusterSchedulableMemoryLow - find the alert in prometheus. click on the query to open it up in the query page and remove `* 85 ` from the end of the queyr to confirm that some data is getting returned.
- ClusterSchedulableCPULow - find the alert in prometheus. click on the query to open it up in the query page and remove `* 85 ` from the end of the queyr to confirm that some data is getting returned.
- MultiAZPodDistribution  - remove `rhoam_version{to_version=""} and ` as this metric is not present running locally and remove `== 1` from the end to show that now data is returned
- ThreeScaleContainerHighMemory - find the alert in prometheus. click on the query to open it up in the query page and remove `> 90` and ensure som date is returned - Note this alert might be firing but this is to be addressed in -> https://issues.redhat.com/browse/MGDAPI-2598
- ThreeScaleContainerHighCPU - find the alert in prometheus. click on the query to open it up in the query page and remove `> 90` and ensure som date is returned

Scripts
Rum tej affected perf test script mentioned in the changes section above in order to  ensure that still works.

Code Base - none of the original queries mentioned above should be found anywhere in the code base.
